README.md

```diff
@@ -47,27 +47,55 @@ Default: `local.net/nfs`
 
 Default: `local-nfs`
 
-### `hostPath`
+### `provisionerVolume`
 
-If this is empty, no local storage will be used, making this completely emphemeral.
+There are a number of modes you can use. These are the available options.
 
-Default: `/srv/nfs-provisioner`
+| mode               | description                                                                                                                                                                                       | available settings                                                          |
+|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| hostPath (default) | A hostPath volume mounts a file or directory from the host node’s filesystem into your pod  https://kubernetes.io/docs/concepts/storage/volumes/#hostpath                                         | * path: "/directory/location/on/host"    (defaults to /srv/nfs-provisioner) |
+| tmpFs              | mount a tmpfs (RAM-backed filesystem) instead of using the host node's storage or a PVC. This is very fast and very volatile but might be useful for things like caches and temporary workspaces. | none                                                                        |
+| emptyDir           | An emptyDir volume is first created when a Pod is assigned to a Node, and exists as long as that Pod is running on that node.  https://kubernetes.io/docs/concepts/storage/volumes/#emptydir      | none                                                                        |
+| pvc                | Make a Persistent Volume Claim for the nfs-provisioner.  https://kubernetes.io/docs/concepts/storage/persistent-volumes/                                                                          | * storageSize: "5Gi" (defaults to 1Gi)                                      |
 
-### `useTmpfs`
+To set a mode, set the `provisionerVolume.mode` value. For instance:
 
-If hostPath is empty and this is set to true, the NFS server will use memory-based
-tmpfs storage instead of allocating disk. This is very fast and very volatile, and
-has the additional risk of consuming cluster memory resources. It will not persist
-across a node restart.
+```console
+--set provisionerVolume.mode=pvc
+```
 
-Default: `false`
+To adjust settings, set the `provisionerVolume.settings.<setting_name>` value. For instance:
+
+```console
+--set provisionerVolume.settings.storageSize=5gi
+```
 
 ### `defaultClass`
 
 Whether to mark this storage provisioner as default. If set to `true`, unlabelled Persistent Volume Claims will use this provisioner.
 
 Default: `false`
 
+### `hostPath`
+
+**REMOVED**
```

Comment:
Is there a way I could throw an error if someone still has these values set? That would ease the transition to the new format.

Reply:
There might be a way to do it using `fail`: http://masterminds.github.io/sprig/flow_control.html

Reply:
Oh great, this is exactly what I was looking for 👍